src/ray/object_manager/plasma/test/mutable_object_test.cc

```diff
@@ -615,6 +615,36 @@ TEST(MutableObjectTest, TestReadMultipleAcquireDuringFailure) {
   }
 }
 
+// Tests that MutableObjectManager instances destruct properly when there are multiple
+// instances.
+// The core worker and the raylet each have their own MutableObjectManager instance, and
+// when both a reader and a writer are on the same machine, the reader and writer will
+// each register the same object with separate MutableObjectManager instances. Thus, we
+// must ensure that the object and its associated metadata (such as the semaphores) are
+// destructed the proper number of times.
+TEST(MutableObjectTest, TestMutableObjectManagerDestruct) {
+  MutableObjectManager manager1;
+  MutableObjectManager manager2;
+  ObjectID object_id = ObjectID::FromRandom();
+  std::string unique_name;
+
+  {
+    std::unique_ptr<plasma::MutableObject> object1 = MakeObject();
+    object1->header->Init();
+    unique_name = std::string(object1->header->unique_name);
+    ASSERT_TRUE(
+        manager1.RegisterChannel(object_id, std::move(object1), /*reader=*/true).ok());
+  }
+  {
+    std::unique_ptr<plasma::MutableObject> object2 = MakeObject();
+    object2->header->Init();
+    memset(object2->header->unique_name, 0, sizeof(object2->header->unique_name));
+    memcpy(object2->header->unique_name, unique_name.c_str(), unique_name.size());
+    ASSERT_TRUE(
+        manager2.RegisterChannel(object_id, std::move(object2), /*reader=*/false).ok());
+  }
```

Comment:
The main thing we're testing here is just that the `manager` destructors don't crash, right? Might be good to add a comment after these blocks explaining that.

Reply:
Done.